src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -111,7 +154,11 @@ private function completeDefinition($id, Definition $definition)
                 // Typehint against a non-existing class
 
                 if (!$parameter->isDefaultValueAvailable()) {
-                    throw new RuntimeException(sprintf('Cannot autowire argument %s for %s because the type-hinted class does not exist (%s).', $index + 1, $definition->getClass(), $reflectionException->getMessage()), 0, $reflectionException);
+                    if ($isConstructor) {
+                        throw new RuntimeException(sprintf('Cannot autowire argument %s for %s because the type-hinted class does not exist (%s).', $index + 1, $definition->getClass(), $reflectionException->getMessage()), 0, $reflectionException);
+                    }
+
+                    return;
```

Comment:
I think this should be a `continue`. It's working now because a setter only ever has 1 argument - but we're in a `foreach`, so the intention is not to exit here - but to keep going